skimage/measure/_polygon.py

```diff
@@ -68,9 +93,9 @@ def approximate_polygon(coords, tolerance):
                               projected_lengths1 > 0)
         eucl = np.logical_not(perp)
         segment_dists[perp] = np.abs(
-            segment_coords[perp, 0] * np.cos(segment_angle)
-            + segment_coords[perp, 1] * np.sin(segment_angle)
-            - segment_dist
+            (segment_coords[perp, 0] * dc - segment_coords[perp, 1] * dr \
```

Comment:
when dr and dc are both 0, euclidean distances are used because the projected lengths become 0.